api/tests/authentication_tests.py

```diff
@@ -0,0 +1,59 @@
+from allauth.socialaccount.models import SocialAccount
+from model_bakery import baker
+import responses
```

Comment:
@say-yawn @jwhitlock how do you like [this `responses` library](https://github.com/getsentry/responses#basics) for mocking `requests`?

Reply:
Oh interesting! You can validate the request data. Looks good to me. Should make testing on APIs be easier too. Was there a lot of places we mocked request? Wondering if we should file a ticket to replace all the old requests or if it's small enough to replace it in this pr.

Reply:
There was even a really cool `assert_call_count` method I used to make sure the `cache` logic was preventing the additional `requests` calls!